pkg/server/api.go

```diff
@@ -186,31 +189,123 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 type healthHandler struct{}
 
-// deriveSpecVersionFromUseragent returns a supported Ignition config spec version for a given Ignition user agent
-// or an error for unsuported user agents.
-// Only Ignition versions that come with support for either spec v2.2 or spec v3.1 are supported
-func deriveSpecVersionFromUseragent(useragent string) (*semver.Version, error) {
-	// for now, serve v2 if we receive a request without a useragent header with Ignition
-	// this happens if the user pings the endpoint directly (e.g. with curl)
-	// and we don't want to break existing behaviour
-	if !strings.HasPrefix(useragent, "Ignition/") {
-		return semver.New("2.2.0"), nil
-	}
-	ignVersionString := strings.SplitAfter(useragent, "/")[1]
-	ignSemver, err := semver.NewVersion(ignVersionString)
-	if err != nil {
-		return ignSemver, errors.Errorf("invalid version in useragent: %s", useragent)
+type acceptHeaderValue struct {
+	MIMEType    string
+	MIMESubtype string
+	Version     *semver.Version
+	QValue      float32
+}
+
+// Parse an accept header, ignoring any extensions that aren't
+// either version or relative quality factor q.
+func parseAcceptHeader(input string) ([]acceptHeaderValue, error) {
+	var header []acceptHeaderValue
+
+	values := strings.Split(input, ",")
+	for i, value := range values {
+		// remove spaces
+		values[i] = strings.TrimSpace(values[i])
+		parts := strings.Split(value, ";")
```

Comment:
It's confusing how we're mutating `values[i]` and then refer to `value` when they're the same thing.  Maybe just:
```
parts := strings.Split(strings.TrimSpace(value), ";")
```
Do we need to mutate `values`?

Reply:
We do not. Fixed!